### ISSUE TITLE
I try to generate transformer MODE_TYPE, but there are some errors...

### ISSUE THREAD

**Issue by mikmeng** [GalacticGlum/composer#11]
Hi Buddy,

I have tried to genenate music_rnn MODEL_TYPE successfully, but after I have trained transformer MODEL_TYPE about two days, I have tried to generate it, but there are the below errors, pls help me, thanks!

(composer) root@fake-server:/opt/composer# composer generate -p ./data/midi/Zukiewicz03.MID  transformer ./output/logdir/transformer-2020-06-16_15-39-40/. ./output/.
2020-06-18 16:11:04.835053: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2020-06-18 16:11:05.407841: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
pciBusID: 0000:21:01.0 name: Tesla T4 computeCapability: 7.5
coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.75GiB deviceMemoryBandwidth: 298.08GiB/s
2020-06-18 16:11:05.408079: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2020-06-18 16:11:05.409581: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2020-06-18 16:11:05.411115: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2020-06-18 16:11:05.411389: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2020-06-18 16:11:05.412949: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2020-06-18 16:11:05.413856: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2020-06-18 16:11:05.417127: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2020-06-18 16:11:05.418582: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0
2020-06-18 16:11:05.418892: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2020-06-18 16:11:05.425478: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2600000000 Hz
2020-06-18 16:11:05.425819: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x555a2f5d9560 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-06-18 16:11:05.425840: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2020-06-18 16:11:05.515690: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x555a2f5db080 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2020-06-18 16:11:05.515723: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5
2020-06-18 16:11:05.516570: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
pciBusID: 0000:21:01.0 name: Tesla T4 computeCapability: 7.5
coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.75GiB deviceMemoryBandwidth: 298.08GiB/s
2020-06-18 16:11:05.516615: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2020-06-18 16:11:05.516630: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2020-06-18 16:11:05.516644: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2020-06-18 16:11:05.516656: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2020-06-18 16:11:05.516669: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2020-06-18 16:11:05.516680: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2020-06-18 16:11:05.516690: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2020-06-18 16:11:05.518066: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0
2020-06-18 16:11:05.518120: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2020-06-18 16:11:05.519222: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-06-18 16:11:05.519240: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 
2020-06-18 16:11:05.519251: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N 
2020-06-18 16:11:05.520689: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14099 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:21:01.0, compute capability: 7.5)
Transformer model restored from './output/logdir/transformer-2020-06-16_15-39-40/./ckpt-1634' (global_step=817000, epoch=9).
  0%|                                                                                                                                                                                                                                                | 0/1024 [00:00<?, ?it/s]2020-06-18 16:11:08.197874: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
  0%|                                                                                                                                                                                                                                                | 0/1024 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/usr/local/anaconda3/envs/composer/bin/composer", line 11, in <module>
    load_entry_point('composer', 'console_scripts', 'composer')()
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/composer/composer/cli.py", line 665, in generate
    predictions = tf.squeeze(predictions, 0)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/util/dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/ops/array_ops.py", line 4197, in squeeze_v2
    return squeeze(input, axis, name)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/util/dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/ops/array_ops.py", line 4145, in squeeze
    return gen_array_ops.squeeze(input, axis, name)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/ops/gen_array_ops.py", line 9861, in squeeze
    input, axis=axis, name=name, ctx=_ctx)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/ops/gen_array_ops.py", line 9897, in squeeze_eager_fallback
    _attr_T, (input,) = _execute.args_to_matching_eager([input], ctx)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/eager/execute.py", line 263, in args_to_matching_eager
    t, dtype, preferred_dtype=default_dtype, ctx=ctx))
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/framework/ops.py", line 1341, in convert_to_tensor
    ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/ops/array_ops.py", line 1455, in _autopacking_conversion_function
    return _autopacking_helper(v, dtype, name or "packed")
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/ops/array_ops.py", line 1391, in _autopacking_helper
    return gen_array_ops.pack(elems_as_tensors, name=scope)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/ops/gen_array_ops.py", line 6337, in pack
    _ops.raise_from_not_ok_status(e, name)
  File "/usr/local/anaconda3/envs/composer/lib/python3.7/site-packages/tensorflow/python/framework/ops.py", line 6653, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: Shapes of all inputs must match: values[0].shape = [1,10,390] != values[1].shape = [8,2,1,16,10,16] [Op:Pack]

**Comment by galacticglum** [comment]
Thanks for bringing this to my attention!

I believe this is due to a mismatch between the output of the ``MusicRNN`` model and the ``Transformer`` model. Whereas MusicRNN only returns a tensor containing the logits for each event in the vocabulary, the transformer model returns a tensor containing the logits, presents, hidden states, and attention weights. 

I've patched it up in commit d3db779, which should *hopefully* fix it 🤞 (note: I haven't tested the commit since I modified the file directly on GitHub. It's possible there is some simple error. Feel free to submit a pull request in that case).

Let me know if there are any other issues.
<hr>

Also, I kindly ask that when submitting issues in the future, that you refrain from posting the full standard output and only include the relevant parts (i.e. the error stacktrace). It would also be helpful to include the command that was ran. It just makes it easier to understand the problem meaning faster response times 🙂
